hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -205,38 +205,40 @@ public HoodieMetadataPayload(Option<GenericRecord> recordOpt) {
   }
 
   protected HoodieMetadataPayload(String key, int type, Map<String, HoodieMetadataFileInfo> filesystemMetadata) {
-    this(key, type, filesystemMetadata, null, null, null, null);
+    this(key, type, filesystemMetadata, null, null, null, null,
+        filesystemMetadata.isEmpty());
```

Comment:
Unfortunately, this is not going to work as `isDeletedRecord` for filesystemMetadata is at the level of `HoodieMetadataFileInfo`. I think `files` partition is very critical. How about we leave this constructor as is and introduce new constructor which instantiates isDeletedRecord for all other metadata payloads.

Reply:
sg.